.circleci/config.yml

```diff
@@ -0,0 +1,53 @@
+version: 2.1
+
+parameters:
+  ubuntu_focal_machine:
+    type: string
+    default: 'ubuntu-2004:202111-02'
+  ubuntu_xenial_machine:
+    type: string
+    default: 'ubuntu-1604:202104-01'
+
+workflows:
+  circleci_build_and_test:
+    jobs:
+      - spin_up:
+          name: 'spin_up_<< matrix.config >>_<< matrix.machine >>'
+          matrix:
+            parameters:
+              config: ['', 'betanet', 'testnet', 'devnet', 'dev', 'beta', 'release', 'nightly']
+              machine: [<< pipeline.parameters.ubuntu_focal_machine >>, << pipeline.parameters.ubuntu_xenial_machine >>]
+
+jobs:
+  spin_up:
+    parameters:
+      config:
+        type: string
+      machine:
+        type: string
+    machine:
+      image: << parameters.machine >>
+    resource_class: medium
+    steps:
+      - checkout
+      - run:
+          name: Install docker
+          command: |
+            export SUDO=sudo
```

Comment:
If you're just sudoing the rest, why not do it explicitly?

Reply:
had to change too many times between having to do sudo, and having not to do sudo, so I just decided to put it up as a flag. I can remove if we want.

Reply:
removed